pkg/serviceaccount/metadata_test.go

```diff
@@ -0,0 +1,240 @@
+	}
+
+	reqURL := s.URL + "/.well-known/openid-configuration"
+
+	resp, err := http.Get(reqURL)
+	if resp == nil || err != nil {
+		t.Errorf("Get(%s) = %v, %v want: <response>, %v", reqURL, resp, err, nil)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("Get(%s) = %v, _ want: %v, _", reqURL, resp.StatusCode, http.StatusOK)
+	}
+
+	if got, want := resp.Header.Get("Content-Type"), "application/json"; got != want {
+		t.Errorf("Get(%s) Content-Type = %q, _ want: %q, _", reqURL, got, want)
+	}
+
+	d := json.NewDecoder(resp.Body)
+	var got Configuration
+	if err := d.Decode(&got); err != nil {
+		t.Errorf("Decode(_) = %v, want: <nil>", err)
+		return // can't evaluate Configuration
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("received configuration differs: got: %+v\nwant: %+v\n", got, want)
+	}
+}
+
+var serveKeysTests = []struct {
+	Name     string
+	Keys     []interface{}
+	WantKeys []jose.JSONWebKey
+}{
+	{
+		Name: "configured public keys",
+		Keys: []interface{}{
+			getPublicKey(rsaPublicKey),
+			getPublicKey(ecdsaPublicKey),
+		},
+		WantKeys: []jose.JSONWebKey{
+			{Key: &rsa.PublicKey{}},
+			{Key: &ecdsa.PublicKey{}},
+		},
+	},
+	{
+		Name: "only publishes public keys",
+		Keys: []interface{}{
+			getPrivateKey(rsaPrivateKey),
+			getPrivateKey(ecdsaPrivateKey),
+		},
+		WantKeys: []jose.JSONWebKey{
+			{Key: &rsa.PublicKey{}},
+			{Key: &ecdsa.PublicKey{}},
+		},
+	},
+}
+
+func TestServeKeys(t *testing.T) {
+	for _, tt := range serveKeysTests {
+		t.Run(tt.Name, func(t *testing.T) {
+			s := setupServer(t, "my-fake-issuer", tt.Keys)
+			defer s.Close()
+
+			reqURL := s.URL + "/serviceaccountkeys/v1/jwks.json"
+
+			resp, err := http.Get(reqURL)
+			if resp == nil || err != nil {
+				t.Errorf("Get(%s) = %v, %v want: <response>, %v", reqURL, resp, err, nil)
+			}
+			if resp.StatusCode != http.StatusOK {
+				t.Errorf("Get(%s) = %v, _ want: %v, _", reqURL, resp.StatusCode, http.StatusOK)
+			}
+			if got, want := resp.Header.Get("Content-Type"), "application/jwk-set+json"; got != want {
+				t.Errorf("Get(%s) Content-Type = %q, _ want: %q, _", reqURL, got, want)
+			}
+			if resp.Body == nil {
+				t.Errorf("resp.Body = %v, want io.ReadCloser", resp.Body)
+				return // can't evaluate body
+			}
+			defer resp.Body.Close()
+
+			d := json.NewDecoder(resp.Body)
+			ks := &jose.JSONWebKeySet{}
+			if err := d.Decode(ks); err != nil {
+				t.Errorf("Decode(_) = %v, want: <nil>", err)
+				return // can't evaluate keyset
+			}
+
+			if got, want := len(ks.Keys), len(tt.WantKeys); got != want {
+				t.Errorf("JWKS: wrong number of keys: got: %d want: %d", got, want)
+			}
+
+			for i := range tt.WantKeys {
+				if i == len(ks.Keys) {
+					return
+				}
+
+				got, want := fmt.Sprintf("%T", ks.Keys[i].Key), fmt.Sprintf("%T", tt.WantKeys[i].Key)
+				if got != want {
+					t.Errorf("Keys[%d]: wrong type: got: %q want: %q", i, got, want)
+				}
+			}
+		})
+	}
+}
+
+func TestDisallowMethods(t *testing.T) {
```

Comment:
Is this test actually useful? Why wouldn't we respond to OPTIONS or HEAD? The CORS filter will respond to OPTIONS regardless.

Reply:
@cceckman any thoughts?

Reply:
Shouldn't what this is trying to test be the responsibility of the server framework, and already be true just by virtue of only registering as a GET?

Reply:
yes, that sounds right to me.

Reply:
yes, that sounds right to me.

Reply:
Yeah, no thoughts / suggestions. Supporting more is fine.

Reply:
Done, removed test